### PR TITLE
[master < T1132] Optimize import of edges on disk

### PR DESCRIPTION
**Pull request by as51340**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/storage/v2/disk/storage.cpp [hunk 1]
```diff
@@ -1233,33 +1247,35 @@ DiskStorage::DiskAccessor::CheckVertexConstraintsBeforeCommit(
 
   /// TODO: andi I don't like that std::optional is used for checking errors but that's how it was before, refactor!
   for (Vertex &vertex : vertex_acc) {
-    if (auto check_result = CheckVertexConstraintsBeforeCommit(vertex, unique_storage); check_result.HasError()) {
-      return check_result.GetError();
-    }
+    if (VertexNeedsToBeSerialized(vertex)) {
```
Comment (Darych) [inline src/storage/v2/disk/storage.cpp:1250]: It is comment more about code style.
In my opinion it's harder to read code when only if statement is inside loop and all logic inside this if statement. Engineer will try to search for else clause or end of if to see if there is something we do outside of if statement.
```
for (Vertex &vertex : vertex_acc) {
  if (VertexNeedsToBeSerialized(vertex)) {
    ...
  }
}
```
Maybe we can rewrite this into:
```
for (Vertex &vertex : vertex_acc) {
  if (!VertexNeedsToBeSerialized(vertex)) {
    continue;
  }
  ...
}
```

Comment (as51340) [reply src/storage/v2/disk/storage.cpp:1249]: I agree but the problem is that iteration over out_edges should happen anyway, that's why I cannot use `continue`

File: src/utils/rocksdb_serialization.hpp [hunk 2]
```diff
@@ -182,9 +182,13 @@ inline std::string SerializeVertexAsValueForLabelIndex(storage::LabelId indexing
   return SerializeVertexAsValueForAuxiliaryStorages(indexing_label, vertex_labels, property_store);
 }
 
-inline std::vector<storage::LabelId> DeserializeLabelsFromLabelIndexStorage(const std::string &value) {
-  const auto value_splitted = value.substr(0, value.find('|'));
-  return TransformFromStringLabels(utils::Split(value_splitted, ","));
+inline storage::LabelId DeserializeLabelFromIndexStorage(const std::string &value) {
+  std::string label = value.substr(0, value.find('|'));
+  return storage::LabelId::FromUint(std::stoull(label));
```
Comment (Darych) [inline src/utils/rocksdb_serialization.hpp:187]: Are we sure that in value we have just one label?

Comment (as51340) [reply src/utils/rocksdb_serialization.hpp:187]: yes, you can check serialization methods for LabelIndex and LabelPropertyIndex

Comment (as51340) [reply src/utils/rocksdb_serialization.hpp:187]: ah Monday bloody Monday... You are right this is for deserializing labels from value string

Comment (as51340) [reply src/utils/rocksdb_serialization.hpp]: and then there was the bug that when we deserialized labels, we didn't include indexing label :exploding_head:

File: src/utils/rocksdb_serialization.hpp [hunk 3]
```diff
@@ -93,7 +95,7 @@ inline std::string SerializeVertexAsValueForAuxiliaryStorages(storage::LabelId l
   std::vector<storage::LabelId> labels_without_target;
   std::copy_if(vertex_labels.begin(), vertex_labels.end(), std::back_inserter(labels_without_target),
                [&label_to_remove](const auto &label) { return label_to_remove != label; });
```
Comment (Darych) [inline src/utils/rocksdb_serialization.hpp:97]: Isn't emplace_back with for loop will be better there also?

Comment (as51340) [reply src/utils/rocksdb_serialization.hpp]: storage::LabelId is trivially copyable so I don't think so but will change

**Review by Darych — APPROVED** [review]

**Review by Darych — APPROVED** [review]

**Review by Darych — APPROVED** [review]

**Review by Darych — APPROVED** [review]

**Pull request merged**